From b7ae8f0a95f8965546135146cc41a0c357c76d68 Mon Sep 17 00:00:00 2001
From: Valentin Raevsky <valentin@compulab.co.il>
Date: Wed, 20 Dec 2023 19:26:05 +0200
Subject: [PATCH 009/155] compulab: imx9: Allow SDP boot w/out hang

Enable CONFIG_ENV_IS_NOWHERE in order to allow passing  board_init_f()
w/out an error. Running w/out this config makes the env_init() return
ENODEV for SDP boot. As a result the board_init_f() runs into hang().

The previous solution was quick and dirt.
Modifying of arch/arm/mach-imx/imx9/soc.c:1032 was a bad tone.
Removed.

Signed-off-by: Valentin Raevsky <valentin@compulab.co.il>
---
 configs/ucm-imx93_defconfig | 1 +
 1 file changed, 1 insertion(+)

diff --git a/configs/ucm-imx93_defconfig b/configs/ucm-imx93_defconfig
index 81fe7ae748b..605f29c8abe 100644
--- a/configs/ucm-imx93_defconfig
+++ b/configs/ucm-imx93_defconfig
@@ -65,6 +65,7 @@ CONFIG_CMD_EXT4_WRITE=y
 CONFIG_OF_CONTROL=y
 CONFIG_SPL_OF_CONTROL=y
 CONFIG_ENV_OVERWRITE=y
+CONFIG_ENV_IS_NOWHERE=y
 CONFIG_SYS_RELOC_GD_ENV_ADDR=y
 CONFIG_ENV_VARS_UBOOT_RUNTIME_CONFIG=y
 CONFIG_NET_RANDOM_ETHADDR=y
-- 
2.34.1

